src/SignalR/common/Protocols.MessagePack/src/Protocol/MessagePackHubProtocolWorker.cs

```diff
@@ -226,7 +226,7 @@ private string[] ReadStreamIds(ref MessagePackReader reader)
 
             if (streamIdCount > 0)
             {
-                streams = new List<string>();
+                streams = new List<string>((int)streamIdCount);
```

Comment:
```suggestion
                streams = new List<string>();
```

We don't want to pre-allocate here for security reasons